src/random.zig

```diff
@@ -129,3 +131,32 @@ pub fn initSeed2D(worldSeed: u64, pos: Vec2i) u64 {
 	const seed = @reduce(.Xor, fac*%pos);
 	return @as(u32, @bitCast(seed)) ^ worldSeed;
 }
+
+pub fn RandomRange(T: type) type {
+	return struct {
+		min: T,
+		max: T,
+
+		pub fn init(min: T, max: T) @This() {
+			return @This(){
+				.min = min,
+				.max = max,
+			};
+		}
+
+		pub fn get(self: @This(), seed: *u64) T {
+			return self.min + (self.max - self.min)*nextFloat(seed);
+		}
+
+		pub fn fromZon(name: []const u8, zon: ZonElement) ?@This() {
+			const vals: ?@Vector(2, T) = if(zon.get(?T, name, null)) |v| @splat(v) else zon.get(?@Vector(2, T), name, null);
```

Comment:
Instead of using zon.get here, and passing the name, I think it would be nicer to have zon.getChild at the callsite and use zon.as here to avoid having to pass the name.